src/components/ImageViewer.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, DOM, createElement } from "react";
+
+import * as Lightbox from "react-image-lightbox";
+
+import { Units } from "./ImageViewerContainer";
+
+import "../ui/ImageViewer.css";
+
+interface ImageViewerProps {
+    imageurl: string;
+    height: number;
+    heightUnits: Units;
+    width: number;
+    widthUnits: Units;
+}
+
+interface ImageViewerState {
+    isOpen: boolean;
+}
+
+class ImageViewer extends Component<ImageViewerProps, ImageViewerState> {
+
+    constructor(props: ImageViewerProps) {
+        super(props);
+
+        this.state = {
+            isOpen: false
+        };
+        this.openLightBox = this.openLightBox.bind(this);
+        this.closeLightBox = this.closeLightBox.bind(this);
+    }
+
+    render() {
+        const divStyle = {
+            height: this.getStyle(this.props.height, this.props.heightUnits),
+            width: this.getStyle(this.props.width, this.props.widthUnits)
+        };
+        const imageStyle = {
+            height: this.getStyle(this.props.height, this.props.heightUnits, true),
+            width: this.getStyle(this.props.width, this.props.widthUnits, true)
+        };
+        return DOM.div({ className: "widget-image-viewer", style: divStyle },
+            DOM.img({
+                onClick: this.openLightBox,
+                src: this.props.imageurl,
+                style: imageStyle
+            }),
+            this.state.isOpen && createElement(Lightbox, {
+                mainSrc: this.props.imageurl,
+                onCloseRequest: this.closeLightBox
+            })
+        );
+    }
+
+    private openLightBox() {
+        this.setState({
+            isOpen: true
+        });
+    }
+
+    private closeLightBox() {
+        this.setState({
+            isOpen: false
+        });
+    }
```

Comment:
Make openLightBox & closeLightBox into single function... can be called toggleLightBoxVisibility or something.